pkg/sql/copy/copy_test: TestCopyFromTransaction failed

pkg/sql/copy/copy_test.TestCopyFromTransaction [failed](https://tanzanite.cluster.engflow.com/invocations/default/3b150448-95fb-4ba7-88b6-da650615b479?testReportRun=3&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NvcHk6Y29weV90ZXN0) on release-24.1.7-rc @ [3613a619e327fb412ac62036ce2c7060c25e48db](https://github.com/cockroachdb/cockroach/commits/3613a619e327fb412ac62036ce2c7060c25e48db):


```
=== RUN   TestCopyFromTransaction
--- FAIL: TestCopyFromTransaction (21.46s)
=== RUN   TestCopyFromTransaction/disableAutoCommitDuringExec=true
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCopyFromTransaction_disableAutoCommitDuringExec=true195681774
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    copy_test.go:303: 
        	Error Trace:	pkg/sql/copy/copy_test/pkg/sql/copy/copy_test.go:303
        	            				pkg/sql/copy/copy_test/pkg/sql/copy/copy_test.go:338
        	Error:      	Received unexpected error:
        	            	ERROR: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/51/1/1/0): "sql txn" meta={id=5f13aeff key=/Tenant/10/Table/3/1/104/2/1 iso=Serializable pri=0.00552554 epo=0 ts=1731673282.655070740,1 min=1731673281.215145263,0 seq=0} lock=true stat=PENDING rts=1731673282.655070740,1 wto=false gul=1731673281.715145263,0 (SQLSTATE 40001)
        	Test:       	TestCopyFromTransaction/disableAutoCommitDuringExec=true
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestCopyFromTransaction_disableAutoCommitDuringExec=true195681774
    --- FAIL: TestCopyFromTransaction/disableAutoCommitDuringExec=true (14.62s)
=== RUN   TestCopyFromTransaction/disableAutoCommitDuringExec=true/explicit_copy
    testing.go:1576: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
        --- FAIL: TestCopyFromTransaction/disableAutoCommitDuringExec=true/explicit_copy (3.50s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCopyFromTransaction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-44424

--- a/pkg/sql/copy/copy_test.go
+++ b/pkg/sql/copy/copy_test.go
@@ -25,6 +25,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/ccl"
 	"github.com/cockroachdb/cockroach/pkg/cli/clisqlclient"
 	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
 	"github.com/cockroachdb/cockroach/pkg/security/username"
 	"github.com/cockroachdb/cockroach/pkg/sql"
@@ -266,6 +267,9 @@ func TestCopyFromTransaction(t *testing.T) {
 		defer srv.Stopper().Stop(ctx)
 
 		s := srv.ApplicationLayer()
+		// Disable pipelining. Without this, pipelined writes performed as part
+		// of the COPY can be lost, which can then cause the COPY to fail.
+		kvcoord.PipelinedWritesEnabled.Override(ctx, &s.ClusterSettings().SV, false)
 
 		url, cleanup := sqlutils.PGUrl(t, s.AdvSQLAddr(), "copytest", url.User(username.RootUser))
 		defer cleanup()

